dev/core/src/com/google/gwt/dev/jjs/impl/GwtAstBuilder.java

```diff
@@ -537,13 +537,25 @@ public void endVisit(BreakStatement x, BlockScope scope) {
 
     @Override
     public void endVisit(CaseStatement x, BlockScope scope) {
+      if (x.isExpr) {
+        InternalCompilerException excption =
+            new InternalCompilerException("Switch expressions not yet supported");
+        excption.addNode(new JCaseStatement(makeSourceInfo(x), null));
+        throw excption;
```

Comment:
```suggestion
        InternalCompilerException exception =
            new InternalCompilerException("Switch expressions not yet supported");
        exception.addNode(new JCaseStatement(makeSourceInfo(x), null));
        throw exception;
```

Reply:
Should we also have a temporary exception in the visit/endVisit for classes to make sure they aren't records, and a test to confirm?

Reply:
If we are throwing exception on visit, do we still need to do the same for endVisit?

Reply:
No, if a visitor throws, it will stop visiting.

I think we need more testing of switch exprs to fail, this doesn't cause an error (since it doesn't have a `case`):
```
    int i = switch(0) {
      default:
        yield 0;
    };
```

Actually I can't get it to fail even with a case:
```
    int i = switch(0) {
      case 1:
        yield 2;
      default:
        yield 7;
    };
```
Maybe we also need a check on the switchstatement? Yeah, confirmed, this only fails using `case ... -> ...` or `default -> ...`.

Reply:
Still need a test, and failing code for something like this:
```java
    int i = switch(1) {
      case 1:
        yield 2;
      default:
        yield 7;
    };
```

I think you need a visit/endVisit for SwitchExpression to catch this - the error thrown above is only good for the new `case Foo ->` syntax.